Add H5_CHECK_MPI_VERSION macro to check for at least MPI version of (version, subversion)

jhendersonHDF opened the pull request:
Also fixes an MPI version check for MPI_Aint_diff usage in parallel compression related code that I noticed after reviewing the selection I/O merge.

jhendersonHDF committed:
Fix MPI version check for MPI_Aint_diff usage

soumagne reviewed (commented):
I think it would be good to use a similar macro as in: https://github.com/HDFGroup/vol-daos/blob/master/src/daos_vol.h#L462

jhendersonHDF commented:
> I think it would be good to use a similar macro as in instead: https://github.com/HDFGroup/vol-daos/blob/master/src/daos_vol.h#L462

Yes I think that's a good idea; let me make that change.

jhendersonHDF committed:
Add H5_CHECK_MPI_VERSION macro

Adds new H5_CHECK_MPI_VERSION macro to check for MPI library version/subversion pair of at least the specified pair
